### PR TITLE
sourcetrail: fix darwin build

### PR DESCRIPTION
**Pull request by midchildan**
###### Motivation for this change
Fix sourcetrail on darwin.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112795` at b88db97d run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/eb59a92e1d746d7b86ba3e61c8178372">sourcetrail</a></li></li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`checkPhase` should probably contain `runHook preCheck` and `runHook postCheck`.

Near pkgs/development/tools/sourcetrail/default.nix:234:3:
```
    |
234 |   checkPhase = ''
    |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/tools/sourcetrail/default.nix:170:3:
```
    |
170 |   installPhase = ''
    |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/tools/sourcetrail/default.nix:264:5:
```
    |
264 |     license = licenses.gpl3;
    |     ^
```
</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112795` at b88db97d run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/7cfb8b916eb2aeb5e742eb0b0372e8f5">sourcetrail</a></li></li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`checkPhase` should probably contain `runHook preCheck` and `runHook postCheck`.

Near pkgs/development/tools/sourcetrail/default.nix:234:3:
```
    |
234 |   checkPhase = ''
    |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/tools/sourcetrail/default.nix:170:3:
```
    |
170 |   installPhase = ''
    |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/tools/sourcetrail/default.nix:264:5:
```
    |
264 |     license = licenses.gpl3;
    |     ^
```
</li>
  </ul>
</details>

**Review by timokau — APPROVED** [review]
Thanks. Sorry for the delay. In the future please [consider](https://mtlynch.io/code-review-love/#6-separate-functional-and-non-functional-changes) a separate commit for formatting changes. It would simplify the review and could also be helpful if we want to revert a commit later.

I will take the liberty to squash & merge because of the fixup commit.

File: pkgs/development/tools/sourcetrail/default.nix [hunk 1]
```diff
@@ -194,20 +252,24 @@ stdenv.mkDerivation rec {
...
+  # FIXME: Tests are disabled on some platforms because of faulty detection
+  # logic for libjvm.so. Should work with manual configuration.
+  doCheck = !stdenv.isDarwin && stdenv.isx86_64;
 
   meta = with lib; {
     homepage = "https://www.sourcetrail.com";
     description = "A cross-platform source explorer for C/C++ and Java";
     platforms = platforms.all;
-    license = licenses.gpl3;
+    license = licenses.gpl3Plus;
```
Comment (timokau) [inline pkgs/development/tools/sourcetrail/default.nix:272]: Should the license also be changed in `python.nix`? Please also make a separate commit for license changes in the future.

(I noticed this in my initial review, but then for some reason missed it when I wanted to comment on it.)

Comment (midchildan) [reply pkgs/development/tools/sourcetrail/default.nix:272]: I missed this, thanks for the tip. I'll include the change in the next update.

**Pull request merged**